src/software/ai/passing/cost_function.cpp

```diff
@@ -91,22 +91,22 @@ double Passing::ratePassShootScore(const Field& field, const Team& enemy_team,
             ->IdealMaxRotationToShootDegrees()
             ->value();
 
-    std::vector<Point> obstacles;
+    std::vector<Circle> obstacles;
     for (const Robot& robot : enemy_team.getAllRobots())
     {
-        obstacles.emplace_back(robot.position());
+        obstacles.emplace_back(Circle(robot.position(), ROBOT_MAX_RADIUS_METERS));
```

Comment:
this seems to be a fairly common operation, perhaps we could define a version of `calcMostOpenDirection` that takes a list of robots instead of `Circle`'s? (_out-of-scope for this PR, so totally optional_)

Reply:
Creates a function version that takes a vector<Robot> instead of a Vector<Circle> to address this.

Reply:
I think @MathewMacDougall is totally right here, why don't we just use `calcBestShot`?

Reply:
I've created a new calcBestShot function that takes a vector of Robots to address this.